filament/backend/src/metal/MetalBuffer.h

```diff
@@ -57,18 +70,29 @@ class TrackedMetalBuffer {
     id<MTLBuffer> get() const noexcept { return mBuffer; }
     operator bool() const noexcept { return bool(mBuffer); }
 
-    static uint64_t getAliveBuffers() { return aliveBuffers; }
+    static uint64_t getAliveBuffers() {
+        uint64_t sum = 0;
+        for (const auto& [_, v] : aliveBuffers) {
+            sum += v;
+        }
+        return sum;
+    }
+
+    static uint64_t getAliveBuffers(Type type) {
+        assert_invariant(type != Type::NONE);
+        return aliveBuffers[type];
+    }
 
 private:
     void swap(TrackedMetalBuffer& other) noexcept {
-        id<MTLBuffer> temp = mBuffer;
-        mBuffer = other.mBuffer;
-        other.mBuffer = temp;
+        std::swap(mBuffer, other.mBuffer);
+        std::swap(mType, other.mType);
     }
 
     id<MTLBuffer> mBuffer;
+    Type mType = Type::NONE;
 
-    static std::atomic<uint64_t> aliveBuffers;
+    static std::unordered_map<TrackedMetalBuffer::Type, std::atomic<uint64_t>> aliveBuffers;
```

Comment:
it's really not ideal to have a hashmap look-up each time. Please considere putting all this behind a compile or runtime flag so we can disable it later, once we figured out the issues. It's probably worth filing a bug so we don't forget.

Reply:
Great points-- for this iteration, I think I can avoid the hash map and do the "lookup" at compile time. And good idea, I'll put this behind a flag.